jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditorDialog.kt

```diff
@@ -0,0 +1,151 @@
+import software.aws.toolkits.resources.message
+
+class QueryEditorDialog(
+    private val project: Project,
+    private val lGroupName: String,
+    private val absoluteTimeSelected: Boolean = false
+
+) : DialogWrapper(project) {
+    constructor(project: Project, logGroupName: String) :
+        this(project = project, lGroupName = logGroupName)
+
+    private val view = QueryEditor(project)
+    private val qQueryingLogGroupApiCall = QueryingLogGroups(project)
+    private val action: OkAction = QueryLogGroupOkAction()
+    private val validator = QueryEditorValidator()
+    init {
+        super.init()
+        title = "Query Log Groups"
+        view.absoluteTimeRadioButton.addActionListener {
+            view.qStartDate.isEnabled = true
+            view.qEndDate.isEnabled = true
+            view.RelativeTimeNumber.isEnabled = false
+            view.RelativeTimeUnit.setEnabled(false)
+        }
+        view.relativeTimeRadioButton.addActionListener {
+            view.qStartDate.isEnabled = false
+            view.qEndDate.isEnabled = false
+            view.RelativeTimeNumber.isEnabled = true
+            view.RelativeTimeUnit.setEnabled(true)
+        }
+        view.queryLogGroupsRadioButton.addActionListener {
+            view.queryBox.isEnabled = true
+            view.querySearchTerm.isEnabled = false
+        }
+        view.searchTerm.addActionListener {
+            view.queryBox.isEnabled = false
+            view.querySearchTerm.isEnabled = true
+        }
+    }
+    // qpanel is the underlying panel for the Query Editor
+    override fun createCenterPanel(): JComponent? = view.qpanel
+    override fun doValidate(): ValidationInfo? = validator.validateEditorEntries(view)
+    override fun getOKAction(): Action = action
+    override fun doCancelAction() {
+        super.doCancelAction()
+    }
+
+    override fun doOKAction() {
+        // Do nothing, close logic is handled separately
+    }
+
+    private fun getRelativeTime(unitOfTime: String, relTimeNumber: Long): StartEndDate {
+
+        val endDate = System.currentTimeMillis() / 1000
+        val startDate: Long = when (unitOfTime) {
+            "Minutes" -> endDate - (relTimeNumber * 60)
+            "Hours" -> endDate - (relTimeNumber * 60 * 60)
+            "Days" -> endDate - (relTimeNumber * 60 * 60 * 24)
+            "Week" -> endDate - (relTimeNumber * 60 * 60 * 24 * 7)
+            else -> 0
+        }
+        return StartEndDate(startDate, endDate)
+    }
+    private fun getAbsoluteTime(startDate: Date, endDate: Date): StartEndDate = StartEndDate(((startDate.time) / 1000), ((endDate.time) / 1000))
+
+    private fun getFilterQuery(searchTerm: String): String = "fields @message, @timestamp | filter @message like '$searchTerm'"
+
+    private fun beginQuerying() {
+        if (!okAction.isEnabled) {
+            return
+        }
+        val funDetails = getFunctionDetails()
+        val queryStartDate: Long
+        val queryEndDate: Long
```

Comment:
don't need to declare types here; do `val queryStartDate = ...` on L93/94 where they're actually being assigned